gqcp/include/ONVBasis/SpinUnresolvedONVBasis.hpp

```diff
@@ -311,6 +315,69 @@ class SpinUnresolvedONVBasis:
 
         const auto& one_op_par = one_op.parameters();
 
+        const auto dim = this->dimension();
+        MatrixRepresentationEvaluationContainer<Representation> ONV_iterator {dim};
+
+        SpinUnresolvedONV onv = this->constructONVFromAddress(0);  // start with ONV with address 0
+
+        for (; !ONV_iterator.isFinished(); ONV_iterator.increment()) {  // loops over all possible ONVs
+            for (size_t e1 = 0; e1 < N; e1++) {                         // loop over electrons that can be annihilated
+
+                // Create an ONVPath for each new ONV.
+                ONVPath<SpinUnresolvedONVBasis> onv_path {*this, onv};
+
+                size_t q = onv.occupationIndexOf(e1);  // retrieve orbital index of the electron that will be annihilated
+
+                // The diagonal values are a result of annihilation-creation on the same orbital index and are thus the same as the initial ONV.
+                if (diagonal_values)
+                    ONV_iterator.addRowwise(ONV_iterator.index, one_op_par(q, q));
+
+                // For the non-diagonal values, we will create all possible matrix elements of the Hamiltonian in the routine below.
+                onv_path.annihilate(q, e1);
+
+                while (!onv_path.isFinished() && onv_path.isOrbitalIndexValid()) {
+
+                    // Find next unoccupied orbital (vertical arc).
+                    onv_path.leftTranslateUntilVertical();
+
+                    // Address after the path has been closed.
+                    size_t address = onv_path.addressAfterCreation(onv_path.orbitalIndex(), onv_path.electronIndex());
+
+                    double h_pq = onv_path.sign() * one_op_par(onv_path.orbitalIndex(), q);
+
+                    // Add the one-electron integral as matrix elements of a Hermitian matrix.
+                    ONV_iterator.addColumnwise(address, h_pq);
+                    ONV_iterator.addRowwise(address, h_pq);
+
+                    // Move orbital index such that other unoccupied orbitals can be found within the loop.
+                    onv_path.addVertical();
```

Comment:
I'm not sure I like this call at the end of the loop, as it creates a path with an invalid state. We'll discuss this later.